cmd/veneur-emit/main.go

```diff
@@ -166,6 +167,24 @@ func main() {
 
 	validateFlagCombinations(passedFlags)
 
+	completion := make(chan int, 1)
+	go func() {
+		doEmission(passedFlags)
+		completion <- 0
+	}()
+
+	select {
+	case status := <-completion:
+		os.Exit(status)
+	case <-time.After(*timeout):
+		logrus.WithField("timeout", timeout.String()).
+			Fatal("Timeout exceeded for emission")
+		os.Exit(1)
```

Comment:
Does `.Fatal()` not trigger a non-zero exit already?

Reply:
Yes! Good eye, I put the log line afterward and didn't think to remove the `Exit`!